### ISSUE TITLE
not support to modify num_samples_processed,  so resume model lead to cost rising due to learning_rate is high at the begging.(not support to save num_samples_processed)

### ISSUE THREAD

**Issue by youngstu** [PaddlePaddle/Paddle#8157]

**Comment by QiJune** [comment]
We provides learning rate decay interface, please refer to the optimizer in image classification demo.

https://github.com/PaddlePaddle/book/blob/feda458a90212ba64d95084a275f8a4ad797ec4a/03.image_classification/train.py#L52-L58

**Comment by youngstu** [comment]
**  lr = learning_rate * pow(1.0 - num_samples_processed / learning_rate_decay_a, learning_rate_decay_b)**
Of course i know, but once the trainning is stopped accidentally at pass-00029 and i want resume to train at pass-00029, num_samples_processd would jump 29 * len(samples) to 0

**Comment by youngstu** [comment]
Maybe you would say that i can modify the learning_rate_a, but num_samples_processed increases linearly and reducing by multilier is invalid. 

**tf.train.exponential_decay(learning_rate, global_step, decay_steps, decay_rate, staircase=False, name=None)**  

This is the tensorflow implemention, for inference. (it is convient and simple)

**Comment by QiJune** [comment]
Yes, exactly. That's a problem.
Since we are now working actively on PaddlePaddle fluid version, we have exposed to set global step in optimizer. Please refer to
https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/v2/fluid/optimizer.py#L37
